### ISSUE TITLE
[Bug]: Update error messages

### ISSUE THREAD

**Issue by chiefmikey** [chiefmikey/gitlang#106]
Currently `user not found` is the only error message

Add unique messages depending on actual failure
`API failure` aka `Connection issue` aka `Try again` etc.
`repo not found: repo-name`
`Timeout` aka `Too many API calls` aka `Too much data lol`

**Comment by Akidane47** [comment]
May I have this dance?

**Comment by chiefmikey** [comment]
@Akidane47 Yes please :) I wasn't expecting anyone to even see these issues so give me a bit to refactor a few things so contributing is more straightforward. I'll assign you when ready, hopefully in the next day or two. Thanks!!

**Comment by chiefmikey** [comment]
It's ready for you @Akidane47, reach out here if you need anything. Thanks!

**Comment by Akidane47** [comment]
My first thought is having two separate search bars or buttons for  searching users and repos? I'm noticing it's having a harder time searching or finding repos than it is with users and it could make the error handling much easier. Style wise I'm not sure how you would want that to look if that's a viable solution. If not, I can just start adding middleware to handle different exceptions and errors

**Comment by chiefmikey** [comment]
I'd like to keep it as a single search input + single button, that may change in the future as #101 and #103 are implemented but ideally it will keep leveraging delimiters in a way that is compatible with url destructuring so that the input `chiefmikey/gitlang` results in the path `https://gitlang.net/chiefmikey/gitlang` and vice versa. 

Do you have any examples of repo searches that are giving you issues? Ultimately searching a repo should be easier since it's the same function as searching a user but instead of listing and compiling languages for _all_ of the user's repos it's just pulling results for the _one_ specified repo in search. So if you're seeing anything off with that there may be a bug to look into.

For the error messages, it's probably easiest to implement responses directly in the server routes for [repos](https://github.com/chiefmikey/gitlang/blob/main/routes/helpers/github/repositories.ts), [languages](https://github.com/chiefmikey/gitlang/blob/main/routes/helpers/github/languages.ts) and the [auth](https://github.com/chiefmikey/gitlang/blob/main/routes/helpers/github/auth.ts), then flow the error handling to send the custom message to the front end. 

Example (variable names aren't accurate, just descriptions):
Input: `chiefmikey/repo-doesnt-exist`
Catch error in [languages](https://github.com/chiefmikey/gitlang/blob/main/routes/helpers/github/languages.ts): if error = no repo => `{ error: ${repo}: Not Found }`
Condition in [server router](https://github.com/chiefmikey/gitlang/blob/main/routes/requests/gitlangRouter.ts): If response.error => `{ error: response.error }`
Condition in [front end query](https://github.com/chiefmikey/gitlang/blob/main/client/routes/requests/languages.ts): If response.data.error => `{ error: response.data.error }`
Condition in [front end router](https://github.com/chiefmikey/gitlang/blob/8382c9a1c4/client/routes/index.ts): If response.error => `{ data: { error: response.error } }`
Condition in [front end submit method](https://github.com/chiefmikey/gitlang/blob/8382c9a1c4/client/src/components/App.svelte#L50): If data.error data = `{ error: data.error }`
Condition in [front end html](https://github.com/chiefmikey/gitlang/blob/8382c9a1c4/client/src/components/App.svelte#L140): #if data.error `<h4>{data.error}</h4>`

If you want to stick with middleware I'll need to do a little refactoring, which is no problem and should probably happen anyways. The front end is hosted as static/serverless on github pages hitting a headless api that handles routes for multiple front end apps stuffed into one aws ec2 instance. The gitlang server code is injected into the instance on deployment which has a condition for serving routes based on what apps are currently injected. I'll copy the [auth-server](https://github.com/chiefmikey/auth-server) koa files into this repo so they can be updated from here and injected with deployment, I'll comment when that's done so you can update your fork. 

Thanks in general too for your ideas, it's really helping me think about the current implementation and what info needs to be documented!

**Comment by chiefmikey** [comment]
Also just fyi when you are ready, I've implemented a beta site that will publish a PR merged with the current production code for live testing at https://beta.gitlang.net to make sure the [auth-server](https://github.com/chiefmikey/auth-server) authentication and api calls are working properly.
